### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by artyomwhite**
[DEMO LINK](https://artyomwhite.github.io/react_movies-list-add-form/)

### CODE REVIEW

File: src/components/NewMovie/NewMovie.tsx [hunk 1]
```diff
@@ -1,50 +1,92 @@
...
+  onAdd: (movie: Movie) => void,
+};
+
+export const NewMovie = ({ onAdd }: Prop) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const [description, setDescription] = useState('');
```
Comment (etojeDenys) [inline src/components/NewMovie/NewMovie.tsx:15]: let's combine it in one useState

```
const [newMovie, setNewMovie] = useState({
  title: '',
  ...
})

```

**Review by etojeDenys — CHANGES_REQUESTED** [review]

File: src/components/NewMovie/NewMovie.tsx (hunk 1 above)
Comment (artyomwhite) [reply src/components/NewMovie/NewMovie.tsx]: done ✅

File: src/components/NewMovie/NewMovie.tsx [hunk 2]
```diff
@@ -1,50 +1,96 @@
...
+        value={newMovie.title}
+        onChange={(value) => setNewMovie({ ...newMovie, title: value })}
         required
       />
 
       <TextField
         name="description"
         label="Description"
-        value=""
+        onChange={(value) => setNewMovie({ ...newMovie, description: value })}
```
Comment (VitaliyBondarenko1982) [inline src/components/NewMovie/NewMovie.tsx:62]: it is redundant to make arrow function in jsx for every TextField
pass just reference on setNewMovie and change proper field with event.target.name
will be necessary to change prop onChange in TextField on such
`React.Dispatch<React.SetStateAction<strong>>` and pass name attribute to input
```suggestion
        onChange={setNewMovie}
```

**Review by VitaliyBondarenko1982 — CHANGES_REQUESTED** [review]
Now I can write each field with empty spaces, button still disabled, but can't to see any error
chrome-capture-2023-6-30

File: src/components/NewMovie/NewMovie.tsx (hunk 1 above)
Comment (VitaliyBondarenko1982) [reply src/components/NewMovie/NewMovie.tsx:15]: why did you make again separate state for each form field?
make as consider in this comment abd was in prev commit
https://github.com/mate-academy/react_movies-list-add-form/pull/1707/files#r1278523947

also fix this comment
https://github.com/mate-academy/react_movies-list-add-form/pull/1707#discussion_r1278570530

File: src/components/TextField/TextField.tsx [hunk 3]
```diff
@@ -24,10 +24,8 @@ export const TextField: React.FC<Props> = ({
   required = false,
   onChange = () => {},
 }) => {
-  // generage a unique id once on component load
   const [id] = useState(() => `${name}-${getRandomDigits()}`);
 
-  // To show errors only if the field was touched (onBlur)
   const [touched, setTouched] = useState(false);
   const hasError = touched && required && !value;
```
Comment (VitaliyBondarenko1982) [inline src/components/TextField/TextField.tsx:30]:
```suggestion
  const hasError = touched && required && !value.trim();
```
this comment not fixed
https://github.com/mate-academy/react_movies-list-add-form/pull/1707#pullrequestreview-1553645900

**Review by VitaliyBondarenko1982 — CHANGES_REQUESTED** [review]
If you have any questions - feel free to ask in fe_chat

**Pull request closed without merging**